vsomeip-sys/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+mod cxx_bridge;
+
+use autocxx::prelude::*; // use all the main autocxx functions
+
+// using autocxx to generate Rust bindings for all these
+include_cpp! {
+    #include "vsomeip/vsomeip.hpp"
+    #include "runtime_wrapper.h"
+    #include "application_wrapper.h"
+    #include "message_wrapper.h"
+    #include "payload_wrapper.h"
+    safety!(unsafe) // see details of unsafety policies described in the 'safety' section of the book
```

Comment:
unsafe? Why?

Reply:
We're wrapping C++ code, which seems to mean unsafe by default.